### PR TITLE
SO-4370: consolidate rf2 property names on index models

### PR DESCRIPTION
**Pull request by cmark**
This PR renames the following index fields to make them consistent with the official RF2 specification and allow API consumers to use RF2 property names when sorting/filtering/selecting content:
* SnomedConcept: `primitive` -> `definitionStatusId`
* SnomedRelationship: `group` -> `relationshipGroup`
* SnomedRefSetMember: `referenceSetId` -> `refsetId`
* SnomedRefSetMember: `targetComponent` -> `targetComponentId`

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRestFixtures.java
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRestFixtures.java:236]: A bunch more work, but these property names could also be replaced with the RF2 constants if they "line up".

**Review by kovacstomi — APPROVED** [review]
LGTM!

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedRelationship.java
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedRelationship.java]: The name of the setter was not updated.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/refset/SnomedReferenceSetMember.java
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/refset/SnomedReferenceSetMember.java]: Same with this setter.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedRefSetMemberIndexEntry.java
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedRefSetMemberIndexEntry.java]: The property name should also be updated here.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedRelationshipIndexEntry.java
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedRelationshipIndexEntry.java]: ⚠️ Two capital letters in exception message: "`Relationship Group number...`"

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/QueryRefSetUpdateRequestBuilder.java
Comment (apeteri) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/QueryRefSetUpdateRequestBuilder.java]: `setModuleId` should use `FIELD_MODULE_ID` from `source`, not `FIELD_REFSET_ID`.

**Review by apeteri — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRestFixtures.java
Comment (cmark) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/SnomedRestFixtures.java:236]: Yeah, I think there was a reason not to do it but now not so sure about it. Since this is not blocking the merge of the PR, let's pick this up after merge.

**Review by apeteri — APPROVED** [review]
👍

**Pull request merged**
